From f17e4bdfd24829563f5372a13f63567d2e9a8c71 Mon Sep 17 00:00:00 2001
From: Haojian Zhuang <haojian.zhuang@linaro.org>
Date: Fri, 22 Apr 2016 17:23:29 +0800
Subject: [PATCH] ANDROID: arm64: add option to build Image-dtb

Some bootloaders couldn't decompress Image.gz-dtb.

Change-Id: I698cd0c4ee6894e8d0655d88f3ecf4826c28a645
Signed-off-by: Haojian Zhuang <haojian.zhuang@linaro.org>
Signed-off-by: John Stultz <john.stultz@linaro.org>
Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>

[AmitP: Folded following android-4.9 commit changes into this patch
        56b70ac2447f ("ANDROID: ARM64: Ignore Image-dtb from git point of view")]
Signed-off-by: Amit Pundir <amit.pundir@linaro.org>
(cherry picked from commit 6acc728a76cafdcc0945c4f0dd56ce521c884c6c)
Signed-off-by: Guenter Roeck <groeck@chromium.org>

[rebase52(rrangel): Context conflict]
Signed-off-by: Raul E Rangel <rrangel@chromium.org>
---
 arch/arm64/Makefile        | 2 +-
 arch/arm64/boot/.gitignore | 1 +
 arch/arm64/boot/Makefile   | 3 +++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 00373c697f1a..a34d1c9e438e 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -164,7 +164,7 @@ Image.%: Image
 zinstall install:
 	$(Q)$(MAKE) $(build)=$(boot) $@
 
-Image.gz-dtb: vmlinux scripts dtbs
+Image-dtb Image.gz-dtb: vmlinux scripts dtbs
 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
 
 PHONY += vdso_install
diff --git a/arch/arm64/boot/.gitignore b/arch/arm64/boot/.gitignore
index eb3551131b1e..34e35209fc2e 100644
--- a/arch/arm64/boot/.gitignore
+++ b/arch/arm64/boot/.gitignore
@@ -1,3 +1,4 @@
 Image
+Image-dtb
 Image.gz
 Image.gz-dtb
diff --git a/arch/arm64/boot/Makefile b/arch/arm64/boot/Makefile
index 5bb65a9e5d13..2c8cb864315e 100644
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -34,6 +34,9 @@ $(obj)/Image: vmlinux FORCE
 $(obj)/Image.bz2: $(obj)/Image FORCE
 	$(call if_changed,bzip2)
 
+$(obj)/Image-dtb: $(obj)/Image $(DTB_OBJS) FORCE
+	$(call if_changed,cat)
+
 $(obj)/Image.gz: $(obj)/Image FORCE
 	$(call if_changed,gzip)
 
-- 
2.23.0.rc1.153.gdeed80330f-goog

